### PR TITLE
System Check #9: More Logging Tests

### PR DESCRIPTION
**Pull request by yashika-batra**
[logging_infrastructure_test.py]
- refers to PR #153 
- tests the log function (and therefore the read/write info.txt functions)
- tests new_log (logs more characters than the max file size and checks multiple files are created)

### CODE REVIEW

File: state_machine/applications/system-check/tests/logging_infrastructure_test.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
+from lib.logging import get_logfile_name, clear_storage, log
+from os import listdir
+
+
+def run(hardware_dict, result_dict):
+
+    # if no SD Card detected, update result dictionary and return
```
Comment (thetazero) [inline state_machine/applications/system-check/tests/logging_infrastructure_test.py:7]: Comments again

File: state_machine/applications/system-check/tests/logging_infrastructure_test.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    print(f"/sd/ directory: {listdir('/sd/')}")
+
+    for i in range(18):
+        msg = 'x' * 100
+        log(msg, 1000)
+
+    logfile1 = get_logfile_name(0)
+    logfile2 = get_logfile_name(1)
+
+    # should create 2 files
```
Comment (thetazero) [inline state_machine/applications/system-check/tests/logging_infrastructure_test.py:25]: Does running listdir create a file???

**Review by thetazero — CHANGES_REQUESTED** [review]
Looks great, comments misleading/unnecessary however

File: state_machine/applications/system-check/tests/logging_infrastructure_test.py (hunk 2 above)
Comment (yashika-batra) [reply state_machine/applications/system-check/tests/logging_infrastructure_test.py:25]: no it doesn't -- the comment is just poorly worded. We want to check that 2 files were created by making sure the filename is in listdir. exists(filename) isn't supported by circuitpython, so this is how i'm checking that 2 files were created

I'll fix the wording here, thanks for bringing it up

File: state_machine/applications/system-check/tests/logging_infrastructure_test.py [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    if not hardware_dict["SDcard"]:
+        result_dict["LoggingInfrastructure_Test"] = (
+            "Cannot test Logging Infrastructure; no SD Card detected", None)
+        return result_dict
+
+    print("Starting logging infrastructure test...")
+    clear_storage()
+    print(f"/sd/ directory: {listdir('/sd/')}")
+
+    # log 1900 characters
```
Comment (thetazero) [inline state_machine/applications/system-check/tests/logging_infrastructure_test.py:22]: 1800 characters

File: state_machine/applications/system-check/tests/logging_infrastructure_test.py [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+    print(f"/sd/ directory: {listdir('/sd/')}")
+    print(f"/sd/logs/ directory: {listdir('/sd/logs/')}")
+
+    # update result dictionary
+    if logfile1_created and logfile2_created:
+        result_string = """Log and new log functions are working;
+files were created successfully."""
+        result_dict["LoggingInfrastructure_Test"] = (result_string, True)
+        print(result_string)
```
Comment (thetazero) [inline state_machine/applications/system-check/tests/logging_infrastructure_test.py:41]: move print after if/else

**Review by thetazero — CHANGES_REQUESTED** [review]
Pretty much set, great work Yashika

**Pull request merged**